src/Standards/Squiz/Sniffs/Functions/FunctionDeclarationArgumentSpacingSniff.php

```diff
@@ -275,18 +312,41 @@ public function processBracket($phpcsFile, $openBracket)
             }
 
             if ($commaToken !== false) {
-                if ($tokens[($commaToken - 1)]['code'] === T_WHITESPACE) {
+                $endOfPreviousParam = $phpcsFile->findPrevious(Tokens::$emptyTokens, ($commaToken - 1), null, true);
+
+                $spaceBeforeComma = 0;
+                if ($tokens[$endOfPreviousParam]['line'] !== $tokens[$commaToken]['line']) {
+                    $spaceBeforeComma = 'newline';
+                } else if ($tokens[($commaToken - 1)]['code'] === T_WHITESPACE) {
+                    $spaceBeforeComma = $tokens[($commaToken - 1)]['length'];
+                }
+
+                if ($spaceBeforeComma !== 0) {
                     $error = 'Expected 0 spaces between argument "%s" and comma; %s found';
                     $data  = [
                         $params[($paramNumber - 1)]['name'],
-                        $tokens[($commaToken - 1)]['length'],
+                        $spaceBeforeComma,
                     ];
 
                     $fix = $phpcsFile->addFixableError($error, $commaToken, 'SpaceBeforeComma', $data);
                     if ($fix === true) {
-                        $phpcsFile->fixer->replaceToken(($commaToken - 1), '');
+                        $phpcsFile->fixer->beginChangeset();
+                        $phpcsFile->fixer->addContent($endOfPreviousParam, ',');
+                        $phpcsFile->fixer->replaceToken($commaToken, '');
+
+                        for ($i = ($commaToken - 1); $tokens[$i]['code'] === T_WHITESPACE; $i--) {
+                            $phpcsFile->fixer->replaceToken($i, '');
+                        }
+
+                        if ($tokens[$i]['code'] === T_COMMENT
+                            && substr($tokens[$i]['content'], -1) === $phpcsFile->eolChar
+                        ) {
+                            $phpcsFile->fixer->replaceToken($i, substr($tokens[$i]['content'], 0, -1));
+                        }
```

Comment:
This doesn't look safe. A comment which starts with `//` or `#` ends with a newline. Removing its newline effectively makes the following line a comment too. I've made a code suggestion for the test case for this change which demonstrates this bug.

Reply:
@fredden Good catch! Thanks for finding that. I've just added a new commit (to be squashed into the original) which should fix this.